src/features/deployment/github.ts

```diff
@@ -0,0 +1,77 @@
+import { WarpCoreConfig, WarpRouteDeployConfig } from '@hyperlane-xyz/sdk';
+import { useMutation } from '@tanstack/react-query';
+import { stringify } from 'yaml';
+import { useToastError } from '../../components/toast/useToastError';
+import { CreatePrBody, CreatePrResponse, GithubIdentity } from '../../types/api';
+import { normalizeEmptyStrings } from '../../utils/string';
+import { useLatestDeployment } from './hooks';
+import { DeploymentType } from './types';
+import { getDeployConfigFilename, getWarpConfigFilename } from './utils';
+import { isSyntheticTokenType } from './warp/utils';
+
+const warpRoutesPath = 'deployments/warp_routes';
+
+export function useCreateWarpRoutePR(onSuccess: () => void) {
+  const { config, result } = useLatestDeployment();
+
+  const { isPending, mutate, mutateAsync, error, data } = useMutation({
+    mutationKey: ['createWarpRoutePr', config, result],
+    mutationFn: (githubInformation: GithubIdentity) => {
+      if (!config.config || config.type !== DeploymentType.Warp) return Promise.resolve(null);
+      if (!result?.result || result.type !== DeploymentType.Warp) return Promise.resolve(null);
+
+      return createWarpRoutePR(config.config, result.result, githubInformation);
+    },
+    retry: false,
+    onSuccess,
+  });
+
+  useToastError(error, 'Error creating PR for Github');
+
+  return {
+    mutate,
+    mutateAsync,
+    error,
+    isPending,
+    data,
+  };
+}
+
+async function createWarpRoutePR(
+  deployConfig: WarpRouteDeployConfig,
+  warpConfig: WarpCoreConfig,
+  githubInformation: GithubIdentity,
+): Promise<CreatePrResponse> {
+  const deployConfigFilename = getDeployConfigFilename(deployConfig);
+  const warpConfigFilename = getWarpConfigFilename(warpConfig);
+  const firstNonSynthetic = Object.values(deployConfig).find((c) => !isSyntheticTokenType(c.type));
+
+  if (!firstNonSynthetic) throw new Error('Token types cannot all be synthetic');
```

Comment:
I think that you can use the WarpDeploySchema from the sdk to have this same misconfig checking